custom_components/better_thermostat/climate.py

```diff
@@ -305,23 +315,23 @@ def _async_startup(*_):
 					if self.night_temp and isinstance(self.night_temp, numbers.Number):
 						_night_temp = float(self.night_temp)
 						# if the night temperature is lower than min_temp, set it to min_temp
-						if _night_temp < self.min_temp:
+						if _night_temp < self._TRV_min_temp:
 							_LOGGER.error(
 								"better_thermostat %s: Night temperature %s is lower than min_temp %s, setting to min_temp",
 								self.name,
 								_night_temp,
-								self.min_temp
+								self._TRV_min_temp
 							)
-							self._target_temp = self.min_temp
+							self._target_temp = self._TRV_min_temp
 						# if the night temperature is higher than the max_temp, set it to max_temp
-						elif _night_temp > self.max_temp:
+						elif _night_temp > self._TRV_max_temp:
 							_LOGGER.warning(
 								"better_thermostat %s: Night temperature %s is higher than max_temp %s, setting to max_temp",
```

Comment:
Please mind debugging output messages on code changes.

This states that the max_temp is what you compare. But actually, you compare the TRV's max temperature - which isn't the config setting max_temp.